ecosystem/sep-0010.md

```diff
@@ -18,12 +18,21 @@ This SEP defines the standard way for clients such as wallets or exchanges to cr
 
 This protocol is a variation of mutual challenge-response, which uses Stellar transactions to encode challenges and responses.
 
-The authentication flow is as follows:
+It involves three parties:
+- A home domain hosting a [SEP-1 stellar.toml](sep-0001.md) containing a `WEB_AUTH_ENDPOINT` value.
+- A web auth endpoint providing the GET and POST operations discussed in this document.
+- A client who will authenticate with a Stellar account.
+
+The discovery flow is as follows:
+
+1. The client retrieves the [SEP-1 stellar.toml](sep-0001.md) from the home domain. The home domain hosts the stellar.toml.
```

Comment:
I took a look at applying this change here, however I believe this is unnecessary because to do so is to respecify SEP-1 which is linked here. SEP-1 describes how a client should retrieve a stellar.toml from a domain. https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0001.md#specification

Reply:
I don't think SEP-1 defines what we need here, specifically there's no reference to RFC3986 => no resolution for `host:port` vs `host` ambiguity. SEP-1 starts with `Given the domain DOMAIN`, which is one step ahead of  where we are in this spec, and there's a good reason for that.  SEP-1 originated from a need to link Stellar account to some metadata which is stored and accessed off-chain. It is not explicitly mentioned in the specification section, but account de facto provides the entrypoint and root context for SEP-1 flow. As far as SEP-1 is concerned, DOMAIN is whatever we find in account's `home_domain` field.

On the other hand, with SEP-10 we want to adopt Stellar keypairs as an authentication method for traditional web, which means that we need to carefully define how to plug Stellar concepts into the well-established framework of existing standards, in particular [HTTP/1.1: Message Syntax and Routing [RFC7230]](https://tools.ietf.org/html/rfc7230) and [HTTP/1.1: Authentication [RFC7235]](https://tools.ietf.org/html/rfc7235). The disposition for SEP-10 is user agent (client) wants to obtain authentication credentials in order to request some representation of the protected resource from the origin server. Resources in HTTP are identified by URI, so one thing we know for sure is that at the start of SEP-10 client is aware of the target URI, but there's no notion of "home domain" or "domain". Without specification every client implementing SEP-10 has to figure out how to convert the URI of the protected resource into the `DOMAIN` input expected by SEP-1, and that's not a good situation from the compatibility angle.

There's also a security angle, because any loosely specified term like home domain in security boundary definition has a potential for all sorts of vulnerabilities and attacks. For example, RFC3986 goes into great details defining the calculation of URIs equality, which if implemented incorrectly may result in compatibility and/or security issues down the road. Anchoring this SEP to existing well-established internet standards increases the odds that developers will rely on the proper library implementations, instead of going with simple but potentially dangerous approach (like comparing URIs as strings).

Maybe I put too much emphasis on this and this SEP will probably never see the level of adoption where things like that will become important. But if we can eliminate potential ambiguity and make the spec a little more formal, why not take this opportunity?

Reply:
+1000 this change, we need to clarify the home domains format so there aren't mismatches of the same service represented by slightly different home domain representations.

A couple things I want to note:
- Add `the` before `client`
- Are we should RFC-3986 is the standard we want to cite? We need to be very careful with this

Reply:
>...one thing we know for sure is that at the start of SEP-10 client is aware of the target URI...
>...every client implementing SEP-10 has to figure out how to convert the URI of the protected resource into the `DOMAIN` input expected by SEP-1...

I'm not sure this is consistent with what I know of some wallets and their implementations. From the limited wallets I've seen if there's a fixed list of anchors within a wallet it's a domain, not a URL. Our stellar.toml clients in some SDKs take a domain as input and build the URL themselves. Examples of this are:
- Go SDK: https://github.com/stellar/go/blob/38aec89/clients/stellartoml/client.go#L14
- JS SDK: https://github.com/stellar/js-stellar-sdk/blob/6a508cb/src/stellar_toml_resolver.ts#L34-L37

I agree there is some room for improvement here, but I don't think it is detrimental to this changes success, and I still think the improvements you're specifying are improvements to SEP-1. SEP-1 is not limited to account-to-toml lookups, it also intended to generally specify how to go from a domain to a stellar.toml. You're correct that the domain is not sufficiently specified in regards to whether it includes a port or not. Practically this is unlikely to matter because production systems consistently use port 443 and HTTPS, or at least not enough for us to address urgently. None of the Stellar TOML clients in SDKs specify ports.

I think we can address this issue independent of SEP-10 v3.0.0.

I have changed the language at this line slightly in e732188 to more explicitly reference SEP-1, rather than casually reference it.